Packages/Tracking/Physical Hands/Runtime/Scripts/PhysicalHandsButtonBase.cs

```diff
@@ -0,0 +1,344 @@
+        [SerializeField, Tooltip("The distance the button can travel when pressed. " +
+            "\n If left a 0, the travel distance will be the y distance from the parent object.")] 
+        private float _buttonTravelDistance = 0f;
+        [SerializeField, Tooltip("Determines whether the button should only be pressed by a hand.")] 
+        protected bool _shouldOnlyBePressedByHand = false;
+        [SerializeField, Tooltip("Specifies which hand(s) can press the button.")] 
+        protected ChiralitySelection _whichHandCanPressButton = ChiralitySelection.BOTH;
+        [SerializeField, Tooltip("Freeze button travel when base object is moving.")]
+        private bool _freezeButtonTravelOnMovement = true;
+        [SerializeField, Tooltip("How fast can the button move before we consider it moving for the sake of freezing button travel on movement")]
+        private Vector3 _buttonVelocityThreshold = new Vector3(0.1f, 0.1f, 0.1f);
+
+
+
+        private const float BUTTON_PRESS_EXIT_THRESHOLD = 0.5f;
+        private bool _isButtonPressed = false;
+        private bool _contactHandPressing = false;
+        private bool _leftHandContacting = false;
+        private bool _rightHandContacting = false;
+        private Rigidbody _pressableObjectRB = null;
+        private Rigidbody _rigidbody = null;
+        private ConfigurableJoint _configurableJoint;
+        private Vector3 _initialButtonPosition = Vector3.zero;
+        private List<GameObject> _objectsContactingButton = new List<GameObject>();
+
+        protected PhysicalHandsButtonHelper _buttonHelper;
+
+        [Space(20)]
+        [Header("Button Events")]
+        [Space(3)]
+        [Header("Button Press")]
+        [Space(3)]
+        public UnityEvent OnButtonPressed;
+        public UnityEvent OnButtonUnPressed;
+        [Header("Button Contact")]
+        [Space(3)]
+        public UnityEvent<ContactHand> OnHandContact;
+        public UnityEvent<ContactHand> OnHandContactExit;
+        [Header("Button Hover")]
+        [Space(3)]
+        public UnityEvent<ContactHand> OnHandHover;
+        public UnityEvent<ContactHand> OnHandHoverExit;
+
+        private void OnEnable()
+        {
+            Initialize();
+        }
+
+        /// <summary>
+        /// Initialize the button.
+        /// </summary>
+        private void Initialize()
+        {
+            if (_pressableObject == null)
+            {
+                Debug.LogError("Pressable object not assigned. Please assign one to use the button.");
+                enabled = false;
+                return;
+            }
+
+            _rigidbody = GetComponent<Rigidbody>();
+            _rigidbody.isKinematic = true;
+            _rigidbody.constraints = RigidbodyConstraints.FreezeAll;
+
+            // Set default travel distance if one is not set
+            if (_buttonTravelDistance == 0)
+            {
+                // leave 1mm so the button does not clip into the base obect (assuming same thickness)
```

Comment:
_Should we_ assume same thickness? Or should we teach users to make the buttons thicker/offset than the bases?

Reply:
solved this by implementing an auto offset mode which is active by default. should work for all common shapes with optional override if your button is odd. works with any different y scale